### PR TITLE
Job Activity History service skeleton

### PR DESCRIPTION
**Pull request by tbak**
Initial implementation of the job activity history component with stubbed GRPC/REST endpoints.

### CODE REVIEW

File: titus-supplementary-component/job-activity-history/src/main/java/com/netflix/titus/supplementary/jobactivity/endpoint/grpc/JobActivityHistoryGrpcServer.java [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+package com.netflix.titus.supplementary.jobactivity.endpoint.grpc;
+
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
+import com.netflix.titus.runtime.endpoint.common.grpc.AbstractTitusGrpcServer;
+import com.netflix.titus.runtime.endpoint.common.grpc.GrpcEndpointConfiguration;
+
+@Singleton
+public class JobActivityHistoryGrpcServer extends AbstractTitusGrpcServer {
```
Comment (andrew-leung) [inline titus-supplementary-component/job-activity-history/src/main/java/com/netflix/titus/supplementary/jobactivity/endpoint/grpc/JobActivityHistoryGrpcServer.java]: Other classes are using just `JobActivity`, not `JobActivityHistory`.

File: titus-supplementary-component/job-activity-history-springboot/src/main/java/com/netflix/titus/supplementary/jobactivity/JobActivityMain.java [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+        return NettyChannelBuilder
+                .forTarget("leader://titusmaster")
+                .nameResolverFactory(new LeaderNameResolverFactory(leaderResolver, configuration.getMasterGrpcPort(), titusRuntime))
+                .usePlaintext(true)
+                .maxHeaderListSize(65536)
+                .build();
+    }
+
+    @Bean
+    public JobActivityHistoryGrpcServer getTaskRelocationGrpcServer(GrpcEndpointConfiguration configuration,
```
[the remark this answers is not in the document]
Comment (tbak) [reply titus-supplementary-component/job-activity-history-springboot/src/main/java/com/netflix/titus/supplementary/jobactivity/JobActivityMain.java]: Fixed

File: titus-supplementary-component/job-activity-history-springboot/src/main/java/com/netflix/titus/supplementary/jobactivity/JobActivityMain.java [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+        CallMetadataResolveComponent.class,
+        TitusMasterConnectorComponent.class,
+        MasterConnectorComponent.class,
+        MasterDataReplicationComponent.class,
+        RestAddOnsComponent.class
+})
+public class JobActivityMain {
+
+    @Bean
+    public LeaderResolver getLeaderResolver(TitusMasterClientConfiguration configuration) {
```
[the remark this answers is not in the document]
Comment (tbak) [reply titus-supplementary-component/job-activity-history-springboot/src/main/java/com/netflix/titus/supplementary/jobactivity/JobActivityMain.java]: Good point. I will create Spring component for this.

File: titus-supplementary-component/job-activity-history/src/main/java/com/netflix/titus/supplementary/jobactivity/endpoint/grpc/JobActivityHistoryGrpcServer.java (hunk 1 above)
Comment (tbak) [reply titus-supplementary-component/job-activity-history/src/main/java/com/netflix/titus/supplementary/jobactivity/endpoint/grpc/JobActivityHistoryGrpcServer.java]: Yes, `JobActivityHistory` is long, and I decided to use `JobActivity` name instead. I will fix it.

**Pull request merged**